Add usage documentations (#258)
stackitcloud/gardener-extension-provider-stackit: docs/usage/usage.md

| `workload-identity.stackit.cloud/service-account-token-expiration-seconds` | `600` | Token lifetime in seconds. |
| `workload-identity.stackit.cloud/idp-token-endpoint` | `https://accounts.stackit.cloud/oauth/v2/token` | Token exchange endpoint. |

Opt out of mutation with: `workload-identity.stackit.cloud/skip-pod-identity-webhook: "true"` label on Pod or Namespace.

### Learn more

For detailed information about STACKIT Workload Identity, see the [STACKIT documentation](https://docs.stackit.cloud/products/runtime/kubernetes-engine/how-tos/workload-identity/).

### References

- [STACKIT Service Account Federation](https://docs.stackit.cloud/platform/access-and-identity/service-accounts/how-tos/manage-service-account-federations/)
- [`stackit-pod-identity-webhook`](https://github.com/stackitcloud/stackit-pod-identity-webhook)
- [Gardener Managed Service Account Issuer](https://gardener.cloud/docs/gardener/security/shoot_serviceaccounts/#managed-service-account-issuer)
- [Kubernetes ServiceAccount token projection](https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken)

> ## STACKIT Application Load Balancer

Reviewer: We should also add a note here that this is only possible if allowed in the admission webhook. We also should document (not in the usage.md, other extensions document such things in [`operations/deployment`](https://github.com/gardener/gardener-extension-provider-gcp/tree/master/docs/operations) ) that this field can be configured. 

https://github.com/stackitcloud/gardener-extension-provider-stackit/blob/14cf9ffe41aa68100c977375c7247225f0fe9007/charts/gardener-extension-admission-stackit/charts/runtime/values.yaml#L24

This extension automatically deploys and configures the STACKIT Application Load Balancer (ALB) Controller in your SKE cluster's control plane, enabling workloads to create Application Load Balancers using Kubernetes `Ingress` resources.

### What this extension provides

- Automatic deployment of the STACKIT ALB Controller in the Shoot control plane

> - Support for creating Application Load Balancers from Kubernetes `Ingress` resources

Reviewer: This is basically the same as the section above and therefore redundant.

> - Deployment of the ALB validating webhook

Reviewer: This does not exist (yet)

> No manual ALB Controller installation is required — the extension handles this for you.

Reviewer: That is nothing we need to mention here. As all "extensions/integrations" work like that.

### Quick start

**Prerequisites:**

- An SKE cluster with the Application Load Balancer extension enabled

> - A Kubernetes `Ingress` resource

Reviewer: This repo has nothing todo with a SKE cluster. It also not an extension from gardener perspective. It is an integration which the gardener-extension-provider-stackit provides. 

Also a `Ingress` resource is not a "requirement". `IngressClass` and `Ingress`  are used to configure ALBs but this is nothing we need to document here (a link to the  Application Load Balancer Controller usage docs are enough.

### Enable the extension

> The Application Load Balancer extension is enabled through the SKE API:

Reviewer: We need to document here how to enable it in the controlplane not how SKE works.

```yaml
extensions:
  applicationLoadBalancer:

>     enabled: true

Reviewer: It is required to have one controller enabled (currently only ingress is supported but gatewayapi will follow)
see validation
https://github.com/stackitcloud/gardener-extension-provider-stackit/blob/a4ef6d0a9dc423ec9c9d5a94c25bce58ff35d172/pkg/apis/stackit/validation/controlplane.go#L83
```suggestion
    enabled: true
    ingress:
      enabled: true
```

```

> Ingress support is enabled automatically when the Application Load Balancer extension is enabled.

Reviewer: No it needs to be enabled.

# Using the STACKIT provider extension with Gardener as end-user

## STACKIT Workload Identity with Gardener

> This extension automatically deploys and configures the `stackit-pod-identity-webhook` in your SKE cluster's control plane, enabling workloads to use STACKIT Workload Identity.

Reviewer: This documentation is about the `garden-extension-provider-stackit` not SKE.

| `workload-identity.stackit.cloud/service-account-token-expiration-seconds` | `600` | Token lifetime in seconds. |
| `workload-identity.stackit.cloud/idp-token-endpoint` | `https://accounts.stackit.cloud/oauth/v2/token` | Token exchange endpoint. |

Opt out of mutation with: `workload-identity.stackit.cloud/skip-pod-identity-webhook: "true"` label on Pod or Namespace.

### Learn more

For detailed information about STACKIT Workload Identity, see the [STACKIT documentation](https://docs.stackit.cloud/products/runtime/kubernetes-engine/how-tos/workload-identity/).

### References

- [STACKIT Service Account Federation](https://docs.stackit.cloud/platform/access-and-identity/service-accounts/how-tos/manage-service-account-federations/)
- [`stackit-pod-identity-webhook`](https://github.com/stackitcloud/stackit-pod-identity-webhook)
- [Gardener Managed Service Account Issuer](https://gardener.cloud/docs/gardener/security/shoot_serviceaccounts/#managed-service-account-issuer)
- [Kubernetes ServiceAccount token projection](https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken)

## STACKIT Application Load Balancer

> This extension automatically deploys and configures the STACKIT Application Load Balancer (ALB) Controller in your SKE cluster's control plane, enabling workloads to create Application Load Balancers using Kubernetes `Ingress` resources.

Reviewer: Not SKE cluster

### Quick start

Refer to [usage doc](https://github.com/stackitcloud/application-load-balancer-controller/blob/main/docs/user.md) of Application Load Balancer for setup. 

### Enable the extension

> The Application Load Balancer extension can be enables by ControlPlaneConfig of the shoot:

Reviewer: ```suggestion
The Application Load Balancer extension can be enabled by ControlPlaneConfig of the shoot:
```

```yaml
extensions:
  applicationLoadBalancer:
    enabled: true
  	ingress:
	  enabled: true
```

> NOTE: Ingress support needs to be enabled in order to use Application Load Balancer extension.

Reviewer: ```suggestion
NOTE: Ingress support must be enabled because at least one source is required. The Application Load Balancer Controller currently only supports Ingress as a source. Support for the Gateway API is planned.
```

### Learn more

For more information about the STACKIT Application Load Balancer Controller, see the [`application-load-balancer-controller`](https://github.com/stackitcloud/application-load-balancer-controller) repository.

# Migrate from Openstack to STACKIT provider

> We still have openstack code runnnig in the repository, and in future it will be fully migrated and use stackit provider.

Reviewer: Lets remove this. There is already a notice in the readme.md
